cli/dev/service.go

```diff
@@ -239,6 +242,36 @@ func ServiceValidate(homeDirectory string, userInputFile string, configFiles []s
 	return absFiles
 }
 
+// Recursively searches dependencies of a service for a particular service URL
+func searchDependencies(dir string, serviceDef *common.ServiceFile, targetService string) (*common.ServiceFile, error) {
+
+	// check the current service
+	if serviceDef.URL == targetService {
+		return serviceDef, nil
+	}
+
+	// generate dependencies
+	serviceDeps, derr := GetServiceDependencies(dir, serviceDef.RequiredServices)
+	if derr != nil {
+		return nil, derr
+	}
+
+	// search dependencies (depth first search)
+	for _, dep := range serviceDeps {
+		res, rerr := searchDependencies(dir, dep, targetService)
+		if rerr != nil {
+			return nil, rerr
+		}
+		if res != nil {
+			return res, nil
+		}
+	}
+
+	// unsuccessful search
+	return nil, nil
+
+}
+
 func ServiceLog(homeDirectory string, serviceName string, tailing bool) {
```

Comment:
This should be changed to 
````
func ServiceLog(homeDirectory string, serviceName string, containerName string, tailing bool)
````
Replacing serviceName with containerName in old code.